component/src/test/java/io/siddhi/extension/io/nats/source/NatsSourceAuthTestCase.java

```diff
@@ -0,0 +1,291 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.siddhi.extension.io.nats.source;
+
+import io.siddhi.core.SiddhiAppRuntime;
+import io.siddhi.core.SiddhiManager;
+import io.siddhi.core.event.Event;
+import io.siddhi.core.stream.output.StreamCallback;
+import io.siddhi.core.util.EventPrinter;
+import io.siddhi.extension.io.nats.utils.NATSClient;
+import io.siddhi.extension.io.nats.utils.ResultContainer;
+import io.siddhi.extension.io.nats.utils.STANClient;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.GenericContainer;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Test case to receive messages while authentication unable.
+ */
+public class NatsSourceAuthTestCase {
+
+    GenericContainer simpleWebServer;
+    private int port = 4222;
+
+    private void startDocker(boolean isTls, String dockerImgName, String... commands) throws InterruptedException {
+        simpleWebServer = new GenericContainer(dockerImgName);
+        simpleWebServer.setPrivilegedMode(true);
+        if (isTls) {
+            simpleWebServer.addFileSystemBind("src/test/resources/certs", "/certs", BindMode.READ_WRITE);
+        }
+        simpleWebServer.setCommand(commands);
+        simpleWebServer.start();
+        port = simpleWebServer.getMappedPort(4222);
+        Thread.sleep(500);
+    }
+
+    @Test
+    public void natsSubscribeWithUsernameAndPassword() throws InterruptedException {
+        startDocker(false, "nats", "--user", "test", "--pass", "1234");
```

Comment:
Do we really need to start a Docker for each test?

Reply:
Same reason as above